content/docs/thinking-in-react.md

```diff
@@ -1,24 +1,24 @@
 ---
 id: thinking-in-react
-title: Thinking in React
+title: Myślenie reaktowe
 permalink: docs/thinking-in-react.html
 redirect_from:
   - 'blog/2013/11/05/thinking-in-react.html'
   - 'docs/thinking-in-react-zh-CN.html'
 prev: composition-vs-inheritance.html
 ---
 
-React is, in our opinion, the premier way to build big, fast Web apps with JavaScript. It has scaled very well for us at Facebook and Instagram.
+Naszym zdaniem React dostarcza pierwszorzędnych narzędzi do budowy dużych szybkich aplikacji webowych. Znakomicie sprawdza się na przykład w naszych zastosowaniach na Facebooku i w Instagramie.
 
-One of the many great parts of React is how it makes you think about apps as you build them. In this document, we'll walk you through the thought process of building a searchable product data table using React.
+Jedną z wielu zalet Reacta jest to jak praca z tą biblioteką uczy cię myśleć o aplikacjach, które tworzysz. Poniżej przybliżymy ci proces myślowy towarzyszący budowie przykładowego programu. Będzie to tabela z danymi o produktach z funkcją wyszukiwania zbudowana w Reakcie.
 
-## Start With A Mock {#start-with-a-mock}
+## Zacznij od projektu {#start-with-a-mock}
 
-Imagine that we already have a JSON API and a mock from our designer. The mock looks like this:
+Załóżmy, że mamy już gotowy JSON API oraz projekt designu. Projekt wygląda następująco:
```

Comment:
```suggestion
Załóżmy, że mamy już gotowe API JSON-owe oraz projekt graficzny. Projekt wygląda następująco:
```

Reply:
Moim zdaniem zależy to głównie od wymowy: *ten* "ej-pi-aj" (jak: *ten* interfejs), ale *to* "api" (jak: *to* okapi).

Reply:
```suggestion
Załóżmy, że mamy już gotowy interfejs JSON API oraz projekt graficzny. Projekt wygląda następująco:
```